packages/telegram-cloud-storage-stamper/src/index.ts

```diff
@@ -0,0 +1,300 @@
+    // check to see if we're in a telegram mini app context
+    this.checkTelegramContext();
+
+    try {
+      return await this.setCloudStorageItem(key, value);
+    } catch {
+      return false;
+    }
+  }
+
+  async insertAPIKey(
+    apiPublicKey: string,
+    apiPrivateKey: string,
+    key: string = DEFAULT_TURNKEY_CLOUD_STORAGE_KEY
+  ) {
+    try {
+      return await this.insertItem(
+        key,
+        this.stringifyAPIKey(apiPublicKey, apiPrivateKey)
+      );
+    } catch {
+      return false;
+    }
+  }
+
+  async getItem(key: string) {
+    // check to see if we're in a telegram mini app context
+    this.checkTelegramContext();
+
+    try {
+      return await this.getCloudStorageItem(key);
+    } catch {
+      return "";
+    }
+  }
+
+  async getAPIKey(key: string = DEFAULT_TURNKEY_CLOUD_STORAGE_KEY) {
+    try {
+      const apiKey = await this.getItem(key);
+
+      if (!apiKey) {
+        return { apiPublicKey: "", apiPrivateKey: "" };
+      }
+
+      return this.parseAPIKey(apiKey as string);
+    } catch {
+      return { apiPublicKey: "", apiPrivateKey: "" };
+    }
+  }
+
+  async clearItem(key: string) {
+    // check to see if we're in a telegram mini app context
+    this.checkTelegramContext();
+
+    try {
+      return await this.clearCloudStorageItem(key);
+    } catch {
+      return false;
+    }
+  }
+
+  checkTelegramContext() {
+    if (window?.Telegram?.WebApp?.CloudStorage == null) {
+      throw new TelegramCloudStorageStamperError(
+        "Cannot use telegram stamper in non telegram mini-app environment, window.Telegram.WebApp.CloudStorage is not defined"
+      );
+    }
+  }
+
+  stringifyAPIKey(apiPublicKey: string, apiPrivateKey: string) {
+    return JSON.stringify({
+      apiPublicKey,
+      apiPrivateKey,
+    });
+  }
+
+  parseAPIKey(apiKey: string) {
+    try {
+      const parsedApiKey = JSON.parse(apiKey);
+
+      if (!this.isApiKey(parsedApiKey)) {
+        return {
+          apiPublicKey: "",
+          apiPrivateKey: "",
+        };
+      }
+
+      return {
+        apiPublicKey: parsedApiKey.apiPublicKey,
+        apiPrivateKey: parsedApiKey.apiPrivateKey,
+      };
+    } catch (err) {
+      throw new TelegramCloudStorageStamperError(
+        "Failed parsing API key from Telegram Cloud Storage"
+      );
+    }
+  }
+
+  isApiKey(apiKey: CloudStorageAPIKey) {
+    return (
+      typeof apiKey.apiPublicKey === "string" &&
+      typeof apiKey.apiPrivateKey === "string"
+    );
+  }
+
+  async getCloudStorageItem(key: string) {
+    return new Promise((resolve, reject) => {
+      window.Telegram.WebApp.CloudStorage.getItem(
+        key,
+        (err: any, value: string) => {
+          if (err != null || !value) {
+            reject(
+              new TelegramCloudStorageStamperError(
+                `Failed getting value: ${key} from Telegram Cloud Storage${
+                  err && `: ${err}`
+                }`
+              )
+            );
+          }
```

Comment:
We need to be a bit careful here, if value is "", this will throw an error. Being able to distinguish between "error" and "empty" cases is important and right now it'd be really hard (devs would have to try/catch and inspect the message).

Instead, let's separate the two?
* If there's an error, `reject` (what you have)
* if there's no error, return `value` (doesn't matter that the value is `""` or undefined IMO, passing what you get out of the TG API will be less confusing than if you tried to transform it)